lib/systems/default.nix

```diff
@@ -83,6 +83,48 @@ let
       # TODO: deprecate args.rustc in favour of args.rust after 23.05 is EOL.
       rust = args.rust or args.rustc or { };
 
+      toolchain = {
+        cc =
+          if final.useLLVM || final.isDarwin then
+            "clang"
+          else if final.useArocc then
+            "arocc"
+          else if final.useZig then
+            "zig"
+          else
+            # TODO: don't imply GCC when nothing else is specified.
+            # Return null once we allow these attributes to be changed.
+            "gcc";
+
+        bintools =
+          if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "llvm" else "gnu";
+
+        cxxlib =
+          if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "libcxx" else "libstdcxx";
+
+        cxxrtlib =
+          if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then
+            "libcxxabi"
+          else if final.isLinux then
+            "libsupc++"
+          else
+            "libcxxrt";
```

Comment:
We don’t use `libcxxrt` on any platform right now (but in the future it will probably make sense to use `libc++` and `libcxxrt` on FreeBSD rather than the current `libstdc++` plus `libsupc++` stack). This should always be `libsupc++` when using `libstdc++`.

Also, we should be consistent around `cxx` vs. `c++` if we diverge from project names. I think I would favour using the upstream project names (so `libstdc++`, `libc++abi`, `libcxxrt`).

Reply:
I've switched to replace `++` with `xx`. The idea behind using `xx` is then it's a bit nicer to interact with directly since you don't have to wrap in a string if it is an attribute name.